src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -90,30 +93,42 @@ public void Unix_ZipWithOSSpecificFileNames(string zipName, string fileName)
             Assert.Equal(fileName, Path.GetFileName(results[0]));
         }
 
-        /// <summary>
-        /// This test ensures that a zipfile with path names that are invalid to this OS will throw errors
-        /// when an attempt is made to extract them.
-        /// </summary>
-        [Theory]
-        [InlineData("NullCharFileName_FromWindows")]
-        [InlineData("NullCharFileName_FromUnix")]
-        [PlatformSpecific(TestPlatforms.Windows)]  // Checks Windows-specific invalid file path
-        public void Windows_ZipWithInvalidFileNames_ThrowsArgumentException(string zipName)
-        {
-            AssertExtensions.Throws<ArgumentException>("path", null, () => ZipFile.ExtractToDirectory(compat(zipName) + ".zip", GetTestFilePath()));
-        }
 
         /// <summary>
-        /// This test ensures that a zipfile with path names that are invalid to this OS will throw errors
-        /// when an attempt is made to extract them.
+        /// This test checks whether or not ZipFile.ExtractToDirectory() is capable of handling filenames
+		/// which contain invalid path characters in Windows.
+        ///  Archive:  InvalidWindowsFileNameChars.zip
+        ///  Test/
+        ///  Test/normalText.txt
+        ///  Test"<>|^A^B^C^D^E^F^G^H^I^J^K^L^M^N^O^P^Q^R^S^T^U^V^W^X^Y^Z^[^\^]^^^_/
+        ///  Test"<>|^A^B^C^D^E^F^G^H^I^J^K^L^M^N^O^P^Q^R^S^T^U^V^W^X^Y^Z^[^\^]^^^_/TestText1"<>|^A^B^C^D^E^F^G^H^I^J^K^L^M^N^O^P^Q^R^S^T^U^V^W^X^Y^Z^[^\^]^^^_.txt
+        ///  TestEmpty/
+        ///  TestText"<>|^A^B^C^D^E^F^G^H^I^J^K^L^M^N^O^P^Q^R^S^T^U^V^W^X^Y^Z^[^\^]^^^_.txt
         /// </summary>
-        [Theory]
-        [InlineData("WindowsInvalid_FromUnix")]
-        [InlineData("WindowsInvalid_FromWindows")]
-        [PlatformSpecific(TestPlatforms.Windows)]  // Checks Windows-specific invalid file path
-        public void Windows_ZipWithInvalidFileNames_ThrowsIOException(string zipName)
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Windows_ZipWithInvalidFileNames()
         {
-            AssertExtensions.Throws<IOException>(() => ZipFile.ExtractToDirectory(compat(zipName) + ".zip", GetTestFilePath()));
+            
+            var testDirectory = GetTestFilePath();
+            ZipFile.ExtractToDirectory(compat("InvalidWindowsFileNameChars.zip"), testDirectory);
+            Assert.True(File.Exists(Path.Combine(testDirectory, "TestText______________________________________.txt")));
```

Comment:
would it be cleaner to extract a local method like

```c#
string CheckExists(testDirectory, path)
{
    Assert.True(File.Exists(Path.Combine(testDirectory, path)));
}
```

Reply:
Would I be placing this method directly into the test file or into testUtilities?

Reply:
Within this method as a local function?

https://devblogs.microsoft.com/premier-developer/dissecting-the-local-functions-in-c-7/

Reply:
There are other places where this is used. Maybe I can put it in test utils and replace it everywhere. Or I can just use local function? Whatever you prefer

Reply:
I don't think it's worth replacing it everywhere. In fact, we can leave the code as you're written it if you like. Not important.